suite-native/atoms/src/Button/IconButton.tsx

```diff
@@ -1,93 +1,98 @@
+    ButtonSize,
+    buttonSchemeToColorsMap,
+    buttonStyle,
+    ButtonStyleProps,
+} from './Button';
+import { Box } from '../Box';
+import { Text } from '../Text';
 
-type IconButtonProps = {
+type IconButtonProps = Omit<PressableProps, 'style'> & {
     iconName: IconName;
-    onPress: () => void;
-    isRounded?: boolean;
-    colorScheme?: ButtonColorScheme;
+    colorScheme?: ButtonColorSchemeName;
     size?: ButtonSize;
     style?: NativeStyleObject;
+    title?: string;
+    isDisabled?: boolean;
 };
 
-type StyleProps = {
-    isRounded: boolean;
-    size: ButtonSize;
-    colorScheme: ButtonColorScheme;
-};
-
-const iconButtonStyle = prepareNativeStyle<StyleProps>(
-    (utils, { isRounded, colorScheme, size }) => {
-        const colorSchemeStyles = {
-            primary: {
-                backgroundColor: utils.colors.green,
-            },
-            gray: {
-                backgroundColor: utils.colors.gray300,
-            },
-            red: {
-                backgroundColor: utils.transparentize(0.9, utils.colors.red),
-            },
-        };
+const iconButtonStyle = mergeNativeStyles([
+    buttonStyle,
+    prepareNativeStyle<ButtonStyleProps>((_, { size, hasTitle }) => {
+        const sizeDimensions = {
+            small: 40,
+            medium: 48,
+            large: 56,
+        } as const satisfies Record<ButtonSize, number>;
 
-        const sizeStyles = {
-            small: {
-                width: 16,
-                height: 16,
-                borderRadius: isRounded ? utils.borders.radii.round : utils.borders.radii.small / 2,
-            },
-            medium: {
-                width: 32,
-                height: 32,
-                borderRadius: isRounded ? utils.borders.radii.round : utils.borders.radii.small,
-            },
-            large: {
-                width: 48,
-                height: 48,
-                borderRadius: isRounded ? utils.borders.radii.round : utils.borders.radii.small,
-            },
-        };
         return {
-            justifyContent: 'center',
-            alignItems: 'center',
-            ...colorSchemeStyles[colorScheme],
-            ...sizeStyles[size],
+            padding: 0,
+            height: sizeDimensions[size],
+            width: hasTitle ? 'auto' : sizeDimensions[size],
         };
-    },
-);
-
-const getIconColor = (colorScheme: ButtonColorScheme) => {
-    switch (colorScheme) {
-        case 'gray':
-            return 'gray700';
-        case 'red':
-            return 'red';
-        default:
-            return 'gray0';
-    }
-};
+    }),
+]);
 
 export const IconButton = ({
     iconName,
-    onPress,
     style,
+    title,
     colorScheme = 'primary',
     size = 'medium',
-    isRounded = false,
+    isDisabled = false,
+    ...pressableProps
 }: IconButtonProps) => {
+    const [isPressed, setIsPressed] = useState(false);
     const { applyStyle } = useNativeStyles();
 
+    const hasTitle = title !== undefined;
```

Comment:
nit: Can be `const hasTitle = !!title`?

Reply:
Sure, I originally had it as you are suggesting. But then I changed it because I wasn't sure if you use this notation. Last time when I converted string to number with `+sringVariable` @Nodonisko asked me not to use it.

`!!value` is okay then?

Reply:
Yes, we use `!!` all over codebase.

Reply:
fixed in 0f7794d